src/workerd/api/streams/internal.c++

```diff
@@ -1656,7 +1656,6 @@ kj::Promise<size_t> IdentityTransformStreamImpl::tryRead(
 
     total += amount;
     buffer = reinterpret_cast<char*>(buffer) + amount;
-    minBytes -= kj::min(minBytes, amount);
```

Comment:
This should fix readAtLeast handling in `IdentityTransformStream`. The read loop was shorting itself by reducing the `minBytes` amount on each iteration.